tutorial/source/baseball.ipynb

```diff
@@ -0,0 +1,3238 @@
+    "\n",
+    "\\begin{align*}\n",
+    "\\mu&\\sim&\\ \\textrm{Normal}(-1, 1)\\\\\n",
+    "\\sigma &\\sim&\\ \\textrm{HalfCauchy}(1)\\\\\n",
+    "\\alpha_i &\\sim&\\ \\textrm{Normal}(\\mu, \\sigma)\\\\\n",
+    "\\phi_i &\\sim&\\ (1 + e^{-\\alpha_i})^{-1} \\\\\n",
+    "h_i &\\sim& \\ \\textrm{Binom}(\\phi_i, K_i)\n",
+    "\\end{align*}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def partially_pooled_with_logit(at_bats, hits=None):\n",
+    "    num_players = at_bats.shape[0]\n",
+    "    loc = pyro.sample(\"loc\", Normal(at_bats.new_tensor(-1), at_bats.new_tensor(1)))\n",
+    "    scale = pyro.sample(\"scale\", HalfCauchy(scale=at_bats.new_tensor(1)))\n",
+    "    alpha = pyro.sample(\"alpha\", Normal(loc, scale).expand_by([num_players]).independent(1))\n",
+    "    return pyro.sample(\"hits\", Binomial(at_bats, logits=alpha), obs=hits)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Model fitting\n",
+    "We are now ready to run our MCMC samplers and examine the performance of the four models. We will walk through the fully pooled model step by step while introducing some new Pyro classes.\n",
+    "\n",
+    "First, we construct and run the `MCMC` object. The first argument to the construction is a NUTS kernel wrapping our model function. We could use an `HMC` kernel instead if we additionally specify a step size and number of steps.\n",
+    "\n",
+    "`MCMC` is a subclass of `TracePosterior`, so after calling `run` the object will contain an `.exec_traces` variable. The entries in this list are the posterior samples generated at each post-warmup step of the Markov chains. A sample is represented by a `Trace` object which contains a node for each `pyro.sample` call in the model function.\n",
+    "\n",
+    "Running this cell will take the majority of the total execution time, proportional to the number of samples in the chains."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "764f0d366ab846f4897fed2ce937e78d",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "SEJveChjaGlsZHJlbj0oSW50UHJvZ3Jlc3ModmFsdWU9MCwgZGVzY3JpcHRpb249dSdXYXJtdXAgWzFdJywgbWF4PTE1MCwgc3R5bGU9UHJvZ3Jlc3NTdHlsZShkZXNjcmlwdGlvbl93aWR0aD3igKY=\n"
```

Comment:
I'm not quite sure why the bar is getting serialized this way for you. I am getting the following. Could you try updating your version of tqdm?

```
      "text/plain": [
       "HBox(children=(IntProgress(value=0, description='Warmup [1]', max=150), HTML(value='')))"
      ]
```

Reply:
The base64 hash appeared in tqdm 4.27 and 4.28 but 4.26 was ok.

Reply:
Hmm..maybe we should just remove our dependency on tqdm. I will work on a simple progress bar implementation and see if I can get it to work well with jupyter.